File: JFFliter/JFFliter/components/JFFliterSectionCell.m
//
//  JFFliterSectionCell.m
//  JFTX_WuHanPJJY
//
//  Created by geeksprite on 2017/7/21.
//  Copyright © 2017年 jf. All rights reserved.
//
#import <Masonry.h>
#import "JFFliterSectionCell.h"
#import "JFFliterAppearanceManager.h"

@interface JFFliterSectionCell ()

@property (nonatomic, strong) UIView *sepLine;

@end

@implementation JFFliterSectionCell

#pragma mark - Init
- (instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier{
    if (self = [super initWithStyle:style reuseIdentifier:reuseIdentifier]) {
        [self didInit];
    }
    return self;
}

- (void)layoutSubviews {
    [super layoutSubviews];
    if (self.width > 0.00001) {
        [self didLayoutSubviews];
    }
}

- (void)didInit {
    self.backgroundColor = [UIColor clearColor];
    [self addSubview:self.sepLine];
}

- (void)didLayoutSubviews {
    CGFloat sepLineHeight = [JFFliterAppearanceManager sharedManager].separateLineStyle == JFFliterSeparateLineStyleNone ? 0.0 : 1.0 ;
    UIColor *sepLineColor = [JFFliterAppearanceManager sharedManager].separateLineStyle == JFFliterSeparateLineStyleBlack ? UIColorWithHEXAndAlpha(@"#030404", 1.0) : UIColorWithHEXAndAlpha(@"#fcfbfb", 1.0);
    self.sepLine.backgroundColor = sepLineColor;
    
    [self.sepLine mas_makeConstraints:^(MASConstraintMaker *make) {
        make.width.bottom.centerX.equalTo(self);
        make.height.mas_equalTo(sepLineHeight);
    }];
}

#pragma mark - Getter && Setter

- (void)setCellItem:(JFFliterItem *)cellItem {
    _cellItem = cellItem;
}

- (UIView *)sepLine {
    if (!_sepLine) {
        _sepLine = [[UIView alloc] init];
    }
    return _sepLine;
}

@end

File: JFFliter/GGSwift/GGSwift/JFFliter/Components/JFFliterSectionCell.swift
//
//  JFFliterSectionCell.swift
//  GGSwift
//
//  Created by geeksprite on 2017/8/1.
//  Copyright © 2017年 聚风天下有限公司. All rights reserved.
//

import UIKit

class JFFliterSectionCell: UITableViewCell {
    
    override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)
        self.didInit()
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func layoutSubviews() {
        super.layoutSubviews()
        guard self.width > 0.00001 else {
            return
        }
        self.didLayoutSubviews()
    }

    open func didInit() {
        self.backgroundColor = UIColor.clear
        guard JFFliterAppearceManager.shared.separateLineStyle != .None else {
            return
        }
        self.sepLine.backgroundColor = JFFliterAppearceManager.shared.separateLineStyle == .Black ? UIColor.colorWithHexString(hex: "#030404") : UIColor.colorWithHexString(hex: "#fcfbfb")
        self.addSubview(self.sepLine)
    }
    
    open func didLayoutSubviews() {
        guard JFFliterAppearceManager.shared.separateLineStyle != .None else {
            return
        }
        let sepHeight : CGFloat = 1.0
        self.sepLine.frame = CGRect.init(x: 0, y: self.height - sepHeight, width: self.width, height: sepHeight)
    }
    
    var cellItem : JFFliterItem? {
        didSet {
    
        }
    }
    
    private lazy var sepLine: UIView = {
        let line = UIView()
        return line
    }()
}
